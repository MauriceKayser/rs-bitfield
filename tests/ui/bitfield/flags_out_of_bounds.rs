//! Assertion failure: _FLAGS_IN_FIELD_0_EXCEED_THE_BIT_FIELD_SIZE

extern crate alloc;

#[bitfield::bitfield(8)] // Can only store flags between 0 - 7.
struct BitField(Flags);

#[derive(Copy, Clone, Debug, bitfield::Flags)]
#[repr(u8)]
enum Flags {
    F0,
    F8 = 8 // This flag is too high to store it in `BitField`.
}

fn main() {}