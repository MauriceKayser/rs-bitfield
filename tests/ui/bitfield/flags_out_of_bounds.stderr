error[E0080]: evaluation of constant value failed
 --> tests/ui/bitfield/flags_out_of_bounds.rs:5:1
  |
5 | #[bitfield::bitfield(8)] // Can only store flags between 0 - 7.
  | ^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at 'Flags in field 0 exceed the bitfield size', $DIR/tests/ui/bitfield/flags_out_of_bounds.rs:5:1
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)
